dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/DockerRegistryAPIHelper.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright 2021 OICR and UCSC
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.dockstore.webservice.helpers;
+
+import static io.dockstore.webservice.Constants.JWT_SECURITY_DEFINITION_NAME;
+
+import com.google.common.hash.Hashing;
+import com.google.gson.Gson;
+import io.dockstore.webservice.CacheHitListener;
+import io.dockstore.webservice.DockstoreWebserviceApplication;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.ws.rs.core.HttpHeaders;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class DockerRegistryAPIHelper {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DockerRegistryAPIHelper.class);
+    private static final Gson GSON = new Gson();
+    private final OkHttpClient client;
+
+    public static final String DOCKER_V2_IMAGE_MANIFEST_MEDIA_TYPE = "application/vnd.docker.distribution.manifest.v2+json";
+    public static final String DOCKER_V2_IMAGE_MANIFEST_LIST_MEDIA_TYPE = "application/vnd.docker.distribution.manifest.list.v2+json";
+    public static final String OCI_IMAGE_MANIFEST_MEDIA_TYPE = "application/vnd.oci.image.manifest.v1+json";
+    public static final String OCI_IMAGE_INDEX_MEDIA_TYPE = "application/vnd.oci.image.index.v1+json";
+
+    public DockerRegistryAPIHelper() {
+        // this code is duplicate from DockstoreWebserviceApplication, except this is a lot faster for unknown reasons ...
+        OkHttpClient.Builder builder = new OkHttpClient().newBuilder();
```

Comment:
We already create two instances, and now this is a third. [Their doc](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/) says it's best to have one for your whole app.

Just checking this pattern is intentional? It sounds like it from the comment.

Reply:
I mainly did it so it would have it's own name for CacheHitListener, but it's not really necessary, so I'll change it to use  DockstoreWebserviceApplication's okhttp client

Reply:
I was partially just asking and I wasn't sure if it was done intentionally... It seemed like @denis-yuen (?) did the existing two on purpose, to have two. So making sure he sees this change.